include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -101,55 +100,160 @@ class BandBlock {
+
+            // First set the diagonals from b+2 to 2b to 0.
+            lapack::laset(General, band_size_ - 1, source.size().cols(), T(0), T(0),
+                          ptr(band_size_ + 1, j), ld() + 1);
+            // The elements are copied in the following way:
+            // (a: copied with first lacpy (General), b: copied with second lacpy (Lower))
+            // 6x6 tile, band_size = 3  |  2x2 tile, band_size = 3
+            // a * * * * *              |
+            // a a * * * *              |  b *
+            // a a a * * *              |  b b
+            // a a a b * *              |
+            // * a a b b *              |
+            // * * a b b b              |
+            const auto index = std::max(SizeType{0}, source.size().cols() - band_size_);
+            if (index > 0) {
+              lapack::lacpy(General, band_size_ + 1, index, source.ptr(), source.ld() + 1, ptr(0, j),
+                            ld() + 1);
+            }
+            const auto size = std::min(band_size_, source.size().cols());
+            lapack::lacpy(Lower, size, size, source.ptr({index, index}), source.ld(), ptr(0, j + index),
+                          ld());
+          },
+          std::move(source));
+    }
+#ifdef DLAF_WITH_CUDA
+    else if constexpr (D == Device::GPU) {
+      DLAF_ASSERT_HEAVY(isAccessibleFromGPU(), "BandBlock memory should be accessible from GPU");
+      return transform(
+          dlaf::internal::Policy<B>(),
+          [=](const matrix::Tile<const T, D>& source, cudaStream_t stream) {
+            constexpr auto General = blas::Uplo::General;
+            constexpr auto Lower = blas::Uplo::Lower;
+
+            // First set the diagonals from b+2 to 2b to 0.
+            gpulapack::laset(General, band_size_ - 1, source.size().cols(), T(0), T(0),
+                             ptr(band_size_ + 1, j), ld() + 1, stream);
+            // The elements are copied in the following way:
+            // (a: copied with first lacpy (General), b: copied with second lacpy (Lower))
+            // 6x6 tile, band_size = 3  |  2x2 tile, band_size = 3
+            // a * * * * *              |
+            // a a * * * *              |  b *
+            // a a a * * *              |  b b
+            // a a a b * *              |
+            // * a a b b *              |
+            // * * a b b b              |
+            const auto index = std::max(SizeType{0}, source.size().cols() - band_size_);
+            if (index > 0) {
+              gpulapack::lacpy(General, band_size_ + 1, index, source.ptr(), source.ld() + 1, ptr(0, j),
+                               ld() + 1, stream);
+            }
+            const auto size = std::min(band_size_, source.size().cols());
+            gpulapack::lacpy(Lower, size, size, source.ptr({index, index}), source.ld(),
+                             ptr(0, j + index), ld(), stream);
+          },
+          std::move(source));
+    }
+#endif
```

Comment:
Commenting only as a reminder to add a link to the issue about removing this duplication.

Reply:
Yes, I'm going to document it today. Thanks for the reminder!